README.md

```diff
@@ -164,5 +164,31 @@ This project can be packaged as either a noarch rpm or an srpm.
 $ make rpm                # Or, `make srpm`
 ```
 
+### i18n
+
+For a general guide on how to write translatable strings, see [weldr.io](http://weldr.io/Translating-welder-web-strings/)
+
+There are a lot of parts involved in translating a string. Here's an overview of the process, from start to finish:
+
+**Step 1**. During development, the developer adds a translatable string. See [weldr.io](http://weldr.io/Translating-welder-web-strings/)
+for details on how to indicate that the string is translatable, and what the string may contain. In general, the string
+is added using [react-intl](https://github.com/yahoo/react-intl) `MessageDescriptor`s, but without explicit `id` attributes.
+
+**Step 2**. The developer runs `node run build`. As part of this process, [babel-plugin-react-intl-auto](https://github.com/akameco/babel-plugin-react-intl-auto)
+will add `id` attributes to all of the messages, and [babel-plugin-react-intl](https://github.com/yahoo/babel-plugin-react-intl)
+will extract all of the messages to JSON files, written to `./build/messages`.
+
+**Step 3**. The developer runs `npm run translations:push`. This uses [react-intl-gettext](https://github.com/rubencosta/react-intl-gettext)
+to collect the JSON files into a gettext-style POT file and uploads the POT file to [Zanata](https://fedora.zanata.org/).
+
+**Step 4**. Translators provide translations on Zanata.
+
+**Step 5**. The developer runs `npm run build`. This downloads the translations from Zanata as gettext-style .po files,
+converts the .po files back to JSON, and bundles the JSON translations with the rest of the project into main.js.
```

Comment:
I'm able to confirm what's happening up until step 2, but is there any way to verify that steps 3 and 5 are actually working as expected? When I go to https://fedora.zanata.org/, is there a way to see what files were uploaded and what translations were provided? The signup process seems to assume that I'm signing up to be a translator, but really I just want to see what's there and what's not, so that I can compare that to what I'm getting when I build.

Do we have any translated strings available yet?

For step 3, does that POT file exist anywhere in the project?

For step 5, am I supposed to be able to see a file somewhere that has the translated text strings?

Reply:
For the `translations:push` target, to upload the .pot file, you will need a Zanata account. And I'll probably need to add you to the project, I can't really remember how that works. You should be able to run `translations:pull` without an account, though, and that will also download the .pot file that's currently on Zanata. Everything is written to ./build/po/

The .pot file is not in the git repo. It's generated from the strings extracted from the javascript files. You can generate the .pot file without doing a push by running `npm run translations:json2pot`. The .pot file will be written to the same place used by `translations:pull`, ./build/po/welder-web.pot.

We have a handful of translated strings. I added some Spanish translations for the simple stuff, and you can see those in ./build/po/es.po, and the .po file is converted to ./build/poJSON/po.es.js. The .js version is a little confusing since the original English strings are not included, but you can see in there, for instance, that "components.Layout.Header.2222696742", which corresponds to a "Help" string, maps to "Ayuda".

Reply:
Thanks for the clarification! I'll create an account for Zanata.

I did switch to Spanish, but never saw any translated strings. But maybe I wasn't looking in the right place.